### PR TITLE
test: fixed test cases after MFE optimization

### PR DESCRIPTION
**Pull request by awais-ansari**
### Description
MFE optimization had done a lot of component updates and restructuring. So many tests are failing now. This PR aims to resolve these components failed test cases. 

- ActionDropdown - Done
- EndorsedAlertBanner - Done
- AlertBanner - Done
- LearnersArea - Done
- LearnersFooter - Done
- LearnerPostsView - Done
- PostLink - Done
- PostFooter - Done
- PostsView - Done

### CODE REVIEW

File: src/discussions/posts/post-editor/PostEditor.test.jsx [hunk 1]
```diff
@@ -110,50 +109,40 @@ describe('PostEditor', () => {
...
+      expect(screen.queryByRole('heading')).toHaveTextContent('Add a post');
+      expect(screen.queryAllByRole('radio')).toHaveLength(2);
+      // 2 categories with 4 subcategories each
+      expect(screen.queryAllByText(/category-\d-topic \d/)).toHaveLength(8);
+      // 3 non courseare topics
+      expect(screen.queryAllByText(/ncw-topic \d/)).toHaveLength(3);
+      expect(screen.queryByText('cohort', { exact: false })).not.toBeInTheDocument();
+
+      if (allowAnonymous) {
+        expect(screen.queryByText('Post anonymously')).not.toBeInTheDocument();
```
Comment (sundasnoreen12) [inline src/discussions/posts/post-editor/PostEditor.test.jsx:126]: it should be expect(screen.queryByText('Post anonymously')).toBeInTheDocument();

**Review by sundasnoreen12 — CHANGES_REQUESTED** [review]

**Review by sundasnoreen12 — CHANGES_REQUESTED** [review]
just small change Remaining LGTM

**Review by sundasnoreen12 — CHANGES_REQUESTED** [review]
just small change Remaining LGTM

**Review by sundasnoreen12 — CHANGES_REQUESTED** [review]
just small change Remaining LGTM

**Review by muhammadadeeltajamul — APPROVED** [review]

**Pull request merged**